aptos-move/aptos-aggregator/src/types.rs

```diff
@@ -212,7 +214,26 @@ impl TryIntoMoveValue for DelayedFieldValue {
         Ok(match (self, layout) {
             (Aggregator(v) | Snapshot(v), U64) => Value::u64(v as u64),
             (Aggregator(v) | Snapshot(v), U128) => Value::u128(v),
-            (Derived(bytes), layout) if is_string_layout(layout) => bytes_to_string(bytes),
+            (Derived(bytes), layout) if is_string_layout(layout) => {
+                if bytes.len() > STRING_AGG_FIXED_LEN {
```

Comment:
So we catch all errors at serialization time, right?

Reply:
Sure, but I'm not landing this on main,  only to preview branch

Reply:
I was just checking if the check here covers all the cases, and it seems so because we always serialize👍